### PR TITLE
MGDCTRS-1921: Update DBZ filter step

### PR DESCRIPTION
**Pull request by indraraj**
Closes https://issues.redhat.com/browse/MGDCTRS-1921
Update the DBZ filter step by

- Placing the Include/Exclude toggle in from the text box
- Update the Filter helper text and step description 
- Removing the bottom divider 
- Add the filter type in the view mode

image
image
image

### CODE REVIEW

File: src/app/pages/CreateConnectorPage/StepConfigurator.tsx [hunk 1]
```diff
@@ -104,6 +104,32 @@ const ConnectedJsonSchemaConfigurator: FunctionComponent<{
   );
 };
 
+export const ConfiguratorCustomStepDescription: FunctionComponent<{
+  actor: ConfiguratorActorRef;
+}> = ({ actor }) => {
+  const { t } = useTranslation();
+  let {
```
Comment (gashcrumb) [inline src/app/pages/CreateConnectorPage/StepConfigurator.tsx:111]: `const` would be better, also if you don't need `configuration` or `connector` you can just delete line 113/114, no need to prepend with `_`.  Then you should also be able to get rid of line 121 and 119, if you don't need those values, you don't need to copy them from the state in the `useSelector` call.

File: src/app/pages/CreateConnectorPage/StepConfigurator.tsx [hunk 2]
```diff
@@ -155,7 +181,13 @@ export const ConfiguratorStep: FunctionComponent = () => {
           ? t(configurationSteps[activeConfigurationStep])
           : t('connectorSpecific')
       }
-      description={t('configurationStepDescription')}
+      description={
+        hasCustomConfigurator ? (
+          <ConfiguratorCustomStepDescription actor={configuratorRef} />
```
Comment (gashcrumb) [inline src/app/pages/CreateConnectorPage/StepConfigurator.tsx:186]: I'd recommend making a custom hook to access the machine and it's state.  [Like here](https://github.com/bf2fc6cc711aee1a0c2a/cos-ui/blob/main/src/app/components/CreateConnectorWizard/CreateConnectorWizardContext.tsx#L321-L322) except you get the `configuratorRef` and then you use `useSelector` to access the active step and then return the right text string.

Comment (indraraj) [reply src/app/pages/CreateConnectorPage/StepConfigurator.tsx]: updated

File: src/app/pages/CreateConnectorPage/StepConfigurator.tsx [hunk 3]
```diff
@@ -33,17 +37,8 @@ const ConnectedCustomConfigurator: FunctionComponent<{
...
-    useCallback(
-      (state: typeof actor.state) => ({
-        connector: state.context.connector,
-        activeStep: state.context.activeStep,
-        configuration: state.context.configuration,
-      }),
-      [actor]
-    )
-  );
+  let { activeStep, configuration, connector } = useConfiguratorMachine();
```
Comment (gashcrumb) [inline src/app/pages/CreateConnectorPage/StepConfigurator.tsx:40]: Excellent!

File: src/utils/shared.ts [hunk 4]
```diff
@@ -195,3 +195,18 @@ export const applyClientSideFormCustomizations = (properties: any) => {
   };
   return organizedProperties;
 };
+
+export const DBFilters: Record<string, string[]> = {
```
Comment (gashcrumb) [inline src/utils/shared.ts:199]: Funny, I was just looking at this `shared.ts` file this morning and was thinking it's getting a little long and unfocused.  I wonder if there's a more specific file we could put shared component helper functions like this?  I think having the shared function is good but in this location it's kind of far from where it's used, at least it seems like it.  Nothing necessary to change for this PR though, just me pontificating :smile:

Comment (indraraj) [reply src/utils/shared.ts:199]: Maybe apart from having global `shared.ts` in `/utils`, we can also create a shared file within each folder(components, machines, pages) catering to those particular sections WDYT?

**Comment by indraraj** [comment]
FYI @jgiardino @nding-anges @MelissaFlinn

**Review by uidoyen — APPROVED** [review]
tested locally. Looks good!

File: src/utils/shared.ts (hunk 4 above)
Comment (gashcrumb) [reply src/utils/shared.ts:199]: Yeah, that sounds like a good idea!  It's at worth trying it out and see how it looks over time.

**Pull request merged**